tests/func.sh

```diff
@@ -357,14 +357,33 @@ check() {
 		max=`cat /proc/sys/dev/raid/speed_limit_max`
 		echo 200000 > /proc/sys/dev/raid/speed_limit_max
 		sleep 0.1
-		while grep -Eq '(resync|recovery|reshape|check|repair) *=' /proc/mdstat ||
-			grep -v idle > /dev/null /sys/block/md*/md/sync_action
-		do
-			sleep 0.5
+		max_iterations=5
+		# wait up to 5 seconds for one of action appeared in sync_action,
+		# which will trigger grow-continue process
+		for ((i = 0 ; i < max_iterations ; i++ )); do
+			sync_action=`grep -Eq '(resync|recovery|reshape|check|repair) *=' /proc/mdstat || grep -v idle > /dev/null /sys/block/md*/md/sync_action`
```

Comment:
Second grep will be executed only if first one fails, is this what we want?

Reply:
It is compatible with the original implementation, I think it should be done this way